src/Gui/CADNavigationStyle.cpp

```diff
@@ -212,7 +211,10 @@ SbBool CADNavigationStyle::processSoEvent(const SoEvent * const ev)
             processed = true;
         }
         else if (this->currentmode == NavigationStyle::DRAGGING) {
-            this->addToLog(event->getPosition(), event->getTime());
+            const auto * const event = (const SoLocation2Event *) ev;
+            if (event) {
```

Comment:
```suggestion
            if (const auto event = dynamic_cast<const SoLocation2Event*>(ev)) {
```

Wouldn't it work as well?

Reply:
Variable **event** is used only in **if (this->currentmode == NavigationStyle::DRAGGING)** so i moved creation to this scope

Reply:
Yes, but if you put the cast into condition you ensure that the `event` is available only in the scope when it also is true.